src/styles/calendar.scss

```diff
@@ -0,0 +1,48 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-size;
+  width: $calendar-size;
+  padding: 10px;
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    box-sizing: border-box;
+    height: $column-size;
+    width: $column-size;
+
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+    background-color: #eee;
+    border: 1px solid black;
+
+    &:hover {
+      transition-property: transform, background-color;
+      transition-duration: 0.5s;
+      transform: translateY(-20px);
+      cursor: pointer;
+      background-color: $hover-color;
+    }
+  }
+
+  @for $day-number from 1 through 31 {
+    & :nth-child(#{$day-number})::before {
+      content: '#{$day-number}';
+    }
+  }
+
+  @each $day, $num in $week-days {
+    &--start-day-#{$day} :first-child {
+      margin-left: ($gap-size * ($num - 1)) + (($num - 1) * $column-size);
```

Comment:
Ensure that the variable `$week-days` is defined in your variables file and follows the Code Style Rules. Also, the calculation should use parentheses to ensure correct order of operations for margin-left value.